.github/workflows/main.yml

```diff
@@ -0,0 +1,45 @@
+name: Test and deploy on target
+
+env:
+  DEPLOY_USER: factory
+  DEPLOY_HOST: c4dtsrv1.epfl.ch
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+
+jobs:
+  build:
+    name: "Test and deploy"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Run `prettier`
+        uses: actionsx/prettier@v2
+        with:
+          args: --check .
+
+      - name: Test
+        uses: fylein/python-pytest-github-action@v2
+        with:
+          args: pip3 install -r requirements.txt && pytest
+
+      - name: Setup ssh key
+        uses: kielabokkie/ssh-key-and-known-hosts-action@v1.2.0
+        if: github.event_name == 'push'
```

Comment:
hum, when pushing a new commit to a pull request, does it trigger the `push` event or only the `pull_request` one? If it does both, then one can trigger a deploy via a PR.

I'm checking [Github documentation](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#push) but it doesn't help that much.

Reply:
Good point, as I remember it was a bit confusing when I looked it up for another project.

My understanding is that it is limited to the branches we define in `on: push: branches:`, and so here this is triggered only when another branch is merged to `main` or if we push directly to `main`.

Reply:
yeah, huum, let's merge it (as we can't use new workflow when not on the main branch) for now, and open/push to a PR to see if it triggers a deploy

Reply:
More precisely, when pushing to a branch, my understanding is that it generates both `push` and `pull_request` events, but the workflow only runs for the `pull_request` event because of the `on: ...` filtering.
But yes let's confirm. :)